libvast/test/module.cpp

```diff
@@ -0,0 +1,689 @@
+//    _   _____   __________
+//   | | / / _ | / __/_  __/     Visibility
+//   | |/ / __ |_\ \  / /          Across
+//   |___/_/ |_/___/ /_/       Space and Time
+//
+// SPDX-FileCopyrightText: (c) 2022 The VAST Contributors
+// SPDX-License-Identifier: BSD-3-Clause
+
+#define SUITE module
+
+#include "vast/module.hpp"
+
+#include "vast/aliases.hpp"
+#include "vast/concept/convertible/data.hpp"
+#include "vast/data.hpp"
+#include "vast/legacy_type.hpp"
+#include "vast/test/test.hpp"
+#include "vast/type.hpp"
+
+#include <caf/error.hpp>
+#include <caf/sum_type.hpp>
+#include <caf/test/dsl.hpp>
+#include <fmt/format.h>
+
+#include <optional>
+#include <string_view>
+
+using namespace vast;
+
+caf::expected<type>
+to_type(const std::vector<type>& known_types, const data& declaration,
+        std::string_view name = "");
```

Comment:
How do we want to name this function?

In order to make this stand out against `type type::infer(const data&)` I suggest we name this `caf::expected<type> type::from_declaration(const data& declaration, ...)`.

Reply:
This should probably come last; let's discuss this with @tobim.

Reply:
Let's do this as part of the YAML modules PR. Resolving for now.